stdlib/src/collections/list.mojo

```diff
@@ -335,6 +335,61 @@ struct List[T: CollectionElement](CollectionElement, Sized):
             earlier_idx += 1
             later_idx -= 1
 
+    @always_inline
+    fn index(self, owned value: T, start: Int = 0) raises -> Int:
+        """Returns the index of the first occurrence of a value in a list; raises ValueError if not found."""
+        var normalized_start = self.size + start if start < 0 else start
+        var normalized_end = self.size
+    
+        if normalized_start >= self.size: raise "Given 'start' parameter is out of range"
+    
+        var ret_val: Int = -1
+        for i in range(normalized_start, normalized_end):
+            # Note: Implementing __contains__ with O(n) time complexity in future, indicating it relies on linear search,
+            # could degrade the performance of the List.index method if it were to replace the current implementation,
+            # as it would essentially perform the same loop twice.
+            if ((self.data + i).bitcast[T]()[]) == value:
+                ret_val = i
+                break
+        else:
+            pass
+            # TODO: Currently, attempting to raise a message, leads to a compiler error (#issue2188)
```

Comment:
FYI issue 2188 seems to be resolved.  Can you uncomment this part and see if the `raise` works?

Reply:
Sure. I'm going to update it.